builtin/logical/aws/client.go

```diff
@@ -123,15 +172,29 @@ func (b *backend) nonCachedClientSTS(ctx context.Context, s logical.Storage, log
 	if err != nil {
 		return nil, err
 	}
-	sess, err := session.NewSession(awsConfig)
-	if err != nil {
-		return nil, err
-	}
-	client := sts.New(sess)
-	if client == nil {
-		return nil, fmt.Errorf("could not obtain sts client")
+
+	var client *sts.STS
+
+	for _, cfg := range awsConfig {
+		sess, err := session.NewSession(cfg)
+		if err != nil {
+			return nil, err
+		}
+		client = sts.New(sess)
+		if client == nil {
+			return nil, fmt.Errorf("could not obtain sts client")
+		}
+
+		// ping the client - we only care about errors
+		_, err = client.GetCallerIdentity(&sts.GetCallerIdentityInput{})
+		if err == nil {
+			return client, nil
+		} else {
+			b.Logger().Debug("couldn't connect with first config, trying next")
```

Comment:
Should we log what region and endpoint failed?

Reply:
good idea